tests/providers/docker/operators/test_docker.py

```diff
@@ -244,7 +244,7 @@ def test_execute_xcom_behavior(self):
         xcom_push_result = xcom_push_operator.execute(None)
         no_xcom_push_result = no_xcom_push_operator.execute(None)
 
-        self.assertEqual(xcom_push_result, b'container log')
+        self.assertEqual(xcom_push_result, 'container log')
```

Comment:
I think this may be a breaking change, WDYT?

Reply:
_run_image of the DockerOperator returns now a python string, not a byte string (otherwise the json encoding fails)
but the unit test did check for a byte string

Reply:
I know, but changing this type is a breaking change. Assuming users used pickle as XCom they were able to pickle that, end then they possibily used `decoded` method which is unavailable for `str` type, so something like this:
```py
the_output_from_xcom.decode("utf-8")
```
won't work, am I right?

Reply:
also: I don't think the `... else line.encode('utf-8')` is correct, 
line is the last element returned from the docker client attach call, according to the airflow documentation xcom should contain the last line if xcom_all is False, but I don't see this happening
see:
https://github.com/docker/docker-py/blob/6da140e26c1cbe8e362b328b1c78d9c736f3a1a2/docker/api/container.py#L17

Reply:
Maybe it's possible to look if [pickling](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#enable-xcom-pickling)  is activated and then return a byte object, otherwise a string.

Reply:
I created an alternative fix, way simpler actually to fix it in the xcom.py at the point of serialization:

https://github.com/apache/airflow/pull/13635

I'd suggest going from there and closing this one

Reply:
@AlessioM I think we should merge this PR which fixes the operator - there's no problem with current XCom implementation.

As per breaking change we can do this - we just need an info about it in ADDITIONAL_INFO.md in docker provider.